### ISSUE TITLE
Create an SQL comparator

### ISSUE THREAD

**Issue by fm-117** [TypeCobolTeam/TypeCobol#2173]
To help create robust unit tests for SQL we need a new SQL comparator. This new comparator will handle "SQL" result files and dump all data from SQL-related code elements and contained SQL objects.

Here is the commit that lead to Symbol comparator creation : https://github.com/TypeCobolTeam/TypeCobol/commit/131441b67774a76aa55ca6f6d010b70d9fbae58a, it can be used as a guide for this new comparator.

- create a new `SqlComparator` class
  - add the diagnostics part
  - while `SymbolComparator` used `ToString` mechanism, we'll use a dedicated visitor along with its Sql visitor
  - exact format is to be determined
- create name classes `SQLName` and `EISQLName`
- add some SQL tests results for existing tests

**Comment by fm-117** [comment]
Ok here is a proposal for SQL comparator format, for example for `ExecSqlWithSelectStar`:

https://github.com/TypeCobolTeam/TypeCobol/blob/843ac2d3dea6e57c494b318e88f396425dcd215d/TypeCobol.Test/Parser/Programs/Cobol85/ExecSql/ExecSqlWithSelectStar.rdz.cbl#L5-L9

We have 3 select statements so the resulting `rdzSQL` file could be :

```
﻿--- Sql Statements ---
Line 6 : SelectStatement
  FullSelect
    SubSelect
      SelectClause
        - SelectionModifier=<NULL>
        - Selections=[
          StarSelection
        ]
Line 7 : SelectStatement
  FullSelect
    SubSelect
      SelectClause
        - SelectionModifier=ALL
        - Selections=[
          StarSelection
        ]
Line 8 : SelectStatement
  FullSelect
    SubSelect
      SelectClause
        - SelectionModifier=DISTINCT
        - Selections=[
          StarSelection
        ]
```

We start with CodeElements with their line to help identify each one in the original code, then each `SqlObject` is on its own line, indented according to its location in the sql objects tree. Properties of a given SqlObject are indented as children but prefixed with '-' to distinguish them from real children.
Arrays or lists are surrounded with brackets and their items are separated with commas.
The absence of value is identified by the constant '<NULL>'.

The idea is to have as much as possible info on the objects created by the parser in a human-readable format.

**Comment by fm-117** [comment]
Closed with #2180.
